core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1149,6 +1149,56 @@ private static RelDataType mapReturnType(SqlOperatorBinding opBinding) {
         false);
   }
 
+  @SuppressWarnings("argument.type.incompatible")
+  private static RelDataType mapKeyReturnType(SqlOperatorBinding opBinding) {
+    List<RelDataType> operandType = new ArrayList<>();
+
+    operandType.add(opBinding.collectOperandTypes().get(0).getKeyType());
+    operandType.add(opBinding.collectOperandTypes().get(0).getValueType());
+    Pair<@Nullable RelDataType, @Nullable RelDataType> type =
+        getComponentTypes(
+            opBinding.getTypeFactory(), operandType);
+
+    requireNonNull(type, "inferred map type");
+    requireNonNull(type.getKey(), "inferred map key type");
+    requireNonNull(type.getValue(), "inferred value type");
+
+    if (!type.getValue().getSqlTypeName().getName().equals("UNKNOWN")
```

Comment:
Compare the `SqlTypeName` enum, not string

Reply:
done, thanks